plugins/inputs/sql/sql.go

```diff
@@ -425,6 +464,21 @@ func (s *SQL) Stop() {
 }
 
 func (s *SQL) Gather(acc telegraf.Accumulator) error {
+	statementsPrepared := true
+	// during plugin startup, it is possible that the server was not reachable.
+	// we try pinging the server in this collection cycle.
+	// we are only concerned with `prepareStatements` function to complete(return true), just once.
+	if !s.serverConnected {
+		if err := s.ping(); err != nil {
+			acc.AddError(err)
+		}
+		statementsPrepared = s.prepareStatements()
+	}
+	// no need to proceed if statements are not prepared
+	if !statementsPrepared {
+		return nil
+	}
```

Comment:
Few thoughts/questions:

1) Do we need the `statementsPrepared` variable with something like the below?
2) prepareStatemnts return is only used when we are not connected, so I think it would be better if that function returned nothing and is only called if we are connected in the first place
3) You make a comment about running `prepareStatements` once, however, my concern is if we ever modify this plugin further and someone changes serverConnected to false somewhere. Then we could re-run `prepareStatemnts` right?

```suggestion
	// during plugin startup, it is possible that the server was not reachable.
	// we try pinging the server in this collection cycle.
	// we are only concerned with `prepareStatements` function to complete(return true), just once.
	if !s.serverConnected {
		if err := s.ping(); err != nil {
			return err
		}
		s.prepareStatements()
	}
```